src/libraries/System.Collections/tests/Generic/Queue/Queue.Generic.Tests.cs

```diff
@@ -203,6 +212,27 @@ public void Queue_Generic_Peek_OnEmptyQueue_ThrowsInvalidOperationException()
 
         #region TrimExcess
 
+        [Theory]
+        [InlineData(1, -1)]
+        [InlineData(2, 1)]
+        public void Queue_TrimAccessWithInvalidArg_ThrowOutOfRange(int size, int newCapacity)
+        {
+            Queue<T> queue = GenericQueueFactory(size);
+
+            AssertExtensions.Throws<ArgumentOutOfRangeException>(() => queue.TrimExcess(newCapacity));
+        }
+
+        [Fact]
+        public void Queue_TrimAccessCurrentCount_DoesNothing()
```

Comment:
As mentioned in earlier discussions, I don't believe this is the correct behaviour. It is reasonable that you might want to trim down the capacity of a collection down to its current size. This is what `Dictionary.TrimExcess(int)` is doing already in .NET 8:

```C#

FieldInfo entriesField = typeof(Dictionary<int, int>).GetField("_entries", BindingFlags.Instance | BindingFlags.NonPublic)!;
Func<Dictionary<int, int>, int> getCapacity = dict => ((Array)entriesField.GetValue(dict)!).Length;

var dict = new Dictionary<int, int>(capacity: 1000) { [0] = 1 };
Console.WriteLine(getCapacity(dict)); // 1103
dict.TrimExcess(dict.Count);
Console.WriteLine(getCapacity(dict)); // 3
```

Reply:
I guess this is just a bad test. The queue implementation is doing exactly you described for dict. The idea for test was `trim to count, and next trim to count will not change capacity again`

Reply:
Yeah, I think it should be changed and renamed so that it checks the behavior of the first trim run.